osf/models/analytics.py

```diff
@@ -136,9 +136,27 @@ def update_counter(cls, page, node_info):
             model_instance.save()
 
     @classmethod
-    def get_basic_counters(cls, page):
+    def _get_file_counter(cls, counter_type, file_id, version=None):
+        """
+        Pulls the total file count for views or downloads across different nodes.
+        :param info_type: String
+        :param file_id: String
+        :param version: Int the version number
+        :return: Int the total count
+        """
+        if version is not None:
+            file_id += ':' + str(version)
+
         try:
-            counter = cls.objects.get(_id=cls.clean_page(page))
-            return (counter.unique, counter.total)
+            return PageCounter.objects.filter(_id__startswith=counter_type,
+                                              _id__endswith=':' + file_id).aggregate(total=models.Sum('total'))['total'] or 0
         except cls.DoesNotExist:
-            return (None, None)
```

Comment:
Since you've changed this query from a `.get()` to a `.filter()`, you should no longer need this `try` `except` block. In fact, pretty sure you don't need the `or 0` part at the end of your aggregate query, as it should return 0 if nothing is found regardless.

Reply:
Aggregating across nothing returns a NoneType not zero, so that is necessary, but the try-catch is not.

Reply:
Strange, when I tried it locally seemed to return 0, but maybe I was doing something different. No biggie either way!

Reply:
Yeah really what's happening is that files with zero views/downloads don't have Pagecounters at all so they will return none, but if you create a Pagecounter object with zero views/downloads and sum over it it will actually return zero, but that shouldn't happen in a real environment as for as I know.